src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -35,8 +34,8 @@ def initialize(server, port, options = {})
 
     @server = server
     @port = port
-    @use_ssl = options[:use_ssl]
-    @protocol = @use_ssl ? "https" : "http"
+    self.class.use_ssl = options[:use_ssl]
```

Comment:
@camlow325 Ah, okay. I mostly wasn't sure about whether or not those settings would always be set. I'll go ahead and fix the tests then, and handle that use case if @cprice404 says it's valid.